pkg/services/dns/dns_config_windows.go

```diff
@@ -0,0 +1,296 @@
+//go:build windows
+
+package dns
+
+import (
+	"errors"
+	"fmt"
+	"net/netip"
+	"strconv"
+	"syscall"
+	"unsafe"
+)
+
+func GetDNSHostAndPort() (string, string, error) {
+	nameservers := getDNSServers()
+
+	var nameserver netip.AddrPort
+	for _, n := range nameservers {
+		// return first non ipv6 nameserver
+		if n.Addr().Is4() {
+			nameserver = n
+			break
+		}
+	}
+
+	return nameserver.Addr().String(), strconv.Itoa(int(nameserver.Port())), nil
+
+}
+
+// copied from https://github.com/qdm12/dns/blob/v2.0.0-beta/pkg/nameserver/getlocal_windows.go
+// this function will use windows syscall to get DNS configuration
```

Comment:
Have you considered using `import "https://github.com/qdm12/dns/v2/pkg/nameserver"` and then calling `nameserver.GetDNSServers()` instead of copying and pasting its code in gvisor-tap-vsock? This would have my preference unless there are good reasons for not doing it.